src/hooks/mousePosition.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from "react";
+
+export const useMousePosition = () => {
```

Comment:
J'aime beaucoup ton idée d'utiliser les hooks. Alors si on go hooks, je te propose de carrément y gérer la position interpolée ! En fait, je me dis qu'on va pouvoir utiliser ton hook pour faire bouger la scene 3d, du coup une seule source de vérité et ça va nous simplifier la vie après. 

 Un sujet très cool qu'on peut taper sur le prochain stream et discuter avec tout le monde. Qu'en dis-tu ?

Ma suggestion en conservant to lerp fait maison : 
```
// Export type to separate types.ts in hooks folder
export type MousePosition = {
  x: number
  y: number
}

export const useMousePosition = () => {
  const [position, setPosition] = useState<MousePosition>({ x: 0, y: 0 })
  const [lerpedPosition, setLerpedPosition] = useState<MousePosition>({
    x: 0,
    y: 0
  })

  useEffect(() => {
    const setFromEvent = (e: MouseEvent) => {
      setPosition({ x: e.clientX, y: e.clientY })
      setTimeout(() => {
        setLerpedPosition({
          x: lerp(position.x, position.x - size / 2, step),
          y: lerp(position.y, position.y - size / 2, step)
        })
      }, 10)
    }

    window.addEventListener("mousemove", setFromEvent)

    return () => {
      window.removeEventListener("mousemove", setFromEvent)
    }
  }, [position])

  return {position, lerpedPosition}
}
```

Reply:
Effectivement, j'approuve l'idée, ça permettrait de profiter du lerp a la fois pour la scène 3d et a la fois pour le point ! Ça serait intéressant d'utiliser les vecteurs three a la place du vieux type mouse position un peu pourri, mais étant donné que three est a utiliser pour de la 3d a la base, ça serait un peu mixer les usages de la lib, a voir si c'est une bonne idée 🤷